frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -125,45 +216,127 @@ const RoutingLabel: React.FC<RoutingLabelProps> = ({ labels }) => {
   return <div className="co-m-label co-m-label--expand">{list}</div>;
 };
 
+const deleteReceiver = (receiverName: string) => {
+  // remove any routes which use receiverToDelete
+  _.update(config, 'route.routes', (routes) => {
+    _.remove(routes, (route: AlertManagerRoute) => route.receiver === receiverName);
+    return routes;
+  });
+  // delete receiver
+  _.update(config, 'receivers', (receivers) => {
+    _.remove(receivers, (receiver: AlertManagerReceiver) => receiver.name === receiverName);
+    return receivers;
+  });
+  return patchAlertManagerConfig(secret, config).then(() => {
+    history.push('/monitoring/alertmanagerconfig');
+  });
+};
+
+const editReceiverMenuItem = (receiverName: string, canUseEditForm: boolean) => {
+  const menuItem = {
+    label: `Edit ${canUseEditForm ? 'Receiver' : 'YAML'}`,
+  };
+  _.set(menuItem, 'callback', () => {
+    const targetUrl = canUseEditForm
+      ? `/monitoring/alertmanagerconfig/receivers/${receiverName}/edit`
+      : `/monitoring/alertmanageryaml`;
+    return history.push(targetUrl);
+  });
+  return menuItem;
+};
+
+const deleteReceiverMenuItem = (
+  receiverName: string,
+  isDefaultReceiver: boolean,
+  canDelete: boolean,
+) => ({
+  label: 'Delete Receiver',
+  isDisabled: isDefaultReceiver || !canDelete,
+  tooltip:
+    isDefaultReceiver || !canDelete
+      ? 'Cannot delete the default receiver, or a receiver which has a sub-route'
+      : '',
+  callback: () =>
+    confirmModal({
+      title: 'Delete Receiver',
+      message: `Are you sure you want to delete receiver '${receiverName}' ?`,
+      btnText: 'Delete Receiver',
+      executeFn: () => deleteReceiver(receiverName),
+    }),
+});
+const receiverMenuActions = (
+  receiverName: string,
+  isDefaultReceiver: boolean,
+  canUseEditForm: boolean,
+  canDelete: boolean,
+) => [
+  editReceiverMenuItem(receiverName, canUseEditForm),
+  deleteReceiverMenuItem(receiverName, isDefaultReceiver, canDelete),
+];
+const ReceiverKebab = ({ receiverName, isDefaultReceiver, canUseEditForm, canDelete }) => (
+  <Kebab
+    options={receiverMenuActions(receiverName, isDefaultReceiver, canUseEditForm, canDelete)}
+  />
+);
```

Comment:
I think that having `ReceiverKebab` and `receiverMenuActions` makes this more verbose and harder to follow.

I'd just use `Kebab` directly and reduce this to

```
const receiverMenuItems = (receiverName: string, canDelete: boolean, canUseEditForm: boolean) => [
  {
    label: `Edit ${canUseEditForm ? 'Receiver' : 'YAML'}`,
    callback: () => {
      const targetUrl = canUseEditForm
        ? `/monitoring/alertmanagerconfig/receivers/${receiverName}/edit`
        : `/monitoring/alertmanageryaml`;
      return history.push(targetUrl);
    },
  },
  {
    label: 'Delete Receiver',
    isDisabled: !canDelete,
    tooltip: !canDelete
      ? 'Cannot delete the default receiver, or a receiver which has a sub-route'
      : '',
    callback: () =>
      confirmModal({
        title: 'Delete Receiver',
        message: `Are you sure you want to delete receiver '${receiverName}' ?`,
        btnText: 'Delete Receiver',
        executeFn: () => deleteReceiver(receiverName),
      }),
  },
];
```

Then in `ReceiverTableRow` you have `<Kebab options={receiverMenuItems(receiver.name, canDelete, canUseEditForm)} />`.

Reply:
agree, much cleaner -thanks